src/main/java/com/ksm/wstbackoffice/service/CountryService.java

```diff
@@ -23,4 +30,35 @@ public CountryService(CountryRepository countryRepository, CountryMapper country
     public CountryDto findById(String id) {
         return countryMapper.toDto(countryRepository.findById(id).orElse(null));
     }
+
+    public Map<String, Map<String, ?>> startWith(Collection<String> nameStartsWithFilters) {
+        Map<String, Map<String, ?>> countriesByNameStartingWith = new HashMap<String, Map<String, ?>>();
+        Map<String, Collection<String>> detailsCountriesByNameStartingWith = new HashMap<String, Collection<String>>();
+        Map<String, Long> totalCountriesByNameStartingWith = new HashMap<String, Long>();
+
+        List<CountryDto> countryDtoList = findAll();
+
+        nameStartsWithFilters.stream()
+                .filter(p -> p.trim().length() > 0)
+                .forEach(p -> {
+                    Set<String> filteredCountries = new HashSet();
+                    countryDtoList.forEach(f -> {
+                        if (f.getName().toLowerCase().startsWith(p.toLowerCase().trim())) {
+                            String name = f.getName();
+                            filteredCountries.add(name);
+                        } else if (f.getName().toLowerCase().split(" ").length > 1) {
+                            Arrays.stream(f.getName().split(" "))
+                                    .filter(e -> e.toLowerCase().startsWith(p.toLowerCase().trim()))
+                                    .map(e -> f.getName()).forEach(filteredCountries::add);
+                        }
+                    });
+                    totalCountriesByNameStartingWith.put(p, filteredCountries.stream().count());
+                    detailsCountriesByNameStartingWith.put(p, filteredCountries);
```

Comment:
What is p variable? Can you name it correctly?

Reply:
done